File: code-coaches/medium/divisible/Divisible.cs
using System;

public class Divisible
{
    static bool IsDivisibleByAll(int num, String[] dividersString)
    {
        foreach (string dividerString in dividersString)
        {
            int divider = int.Parse(dividerString);
            if (num % divider != 0)
            {
                return false;
            }
        }

        return true;
    }

    static void Main(string[] args)
    {
        int num = int.Parse(Console.ReadLine());
        string[] dividersString = Console.ReadLine().Split(' ');

        if (IsDivisibleByAll(num, dividersString))
            Console.WriteLine("divisible by all");
        else
            Console.WriteLine("not divisible by all");
    }
}

File: code-coaches/medium/divisible/divisible.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

bool is_divisible_by_all(int num, int *dividers, int n)
{
    for (int i = 0; i < n; i++)
    {
        if (num % dividers[i] != 0)
        {
            return false;
        }
    }
    return true;
}

int main()
{
    int num;
    scanf("%d", &num);

    char dividers_str[100];
    scanf("%s", dividers_str);

    char* token = strtok(dividers_str, " ");
    int dividers[100];

    int i = 0;
    while (token != NULL && i < 100) // check for NULL and fixed size
    {
        dividers[i] = atoi(token);
        i++;
        token = strtok(NULL, " ");
    }

    if (is_divisible_by_all(num, dividers, i))
        printf("divisible by all");
    else
        printf("not divisible by all");

    return 0;
}
